textbook/2_grouping.qmd

```diff
@@ -0,0 +1,994 @@
+ggplot(income_df, aes(
+    x=percentage_heavy_drinkers, 
+    y=life_expectancy_at_50, 
+    shape=region,
+    color=region
+)) + 
+geom_point()
+```
+
+:::
+
+While this combined approach makes grouping clearer, there are some pitfalls. For instance, line icons limit the total area that can be colored, reducing the impact of color. It is better to use solid shapes when combining shape and color.
+
+Additionally, size matters more than you'd think. Smaller shapes are difficult to distinguish. When plotting a dense scatterplot of small marks, try restricting the number of unique shapes to three or fewer, or consider avoiding using shape altogether.
+
+
+### Size
+
+Size is often used to represent the magnitude of quantitative data. But it can under limited circumstances be used for grouping. Size has an inherent direction, so it works better for **ordinal** categories - those with a clear order. 
+
+One example is World Bank income level, ranging from low to high. Let's shake up our life expectancy chart, encoding the income level of countries in sub-Saharan Africa as size:
+
+::: {.panel-tabset}
+
+# Altair
+
+Here we use `size='income_group'`:
+
+```{python}
+income_df_subset = income_df[income_df['sub-region'].isin(['Sub-Saharan Africa'])]
+
+alt.Chart(income_df_subset).mark_point().encode(
+    x='percentage_heavy_drinkers',
+    y='life_expectancy_at_50',
+    size='income_group',
+)
+```
+
+# ggplot
+
+Here we use `size=income_group`:
+
+```{r fig.width=8, fig.height=6}
+
+income_df_subset <- subset(income_df, `sub-region` %in% c('Sub-Saharan Africa'))
+
+# Create the scatter plot
+ggplot(income_df_subset, aes(
+    x=percentage_heavy_drinkers, 
+    y=life_expectancy_at_50, 
+    size=income_group,
+)) + 
+geom_point()
+
+```
+
+:::
+
+::: {.callout-caution}
+In general, size-based groupings are not very intuitive. Restricting size to four or fewer levels is a good rule of thumb, and for most visualizations grouping by size should be avoided.
+:::
+
+### Position
+
+Position is a powerful grouping strategy. It works by encoding our categorical variable as the x or y axis. Each group has its own visual lane, making it easier to compare overall distributions across categories. It  also emphasizes the outliers within each category.
+
+Below, we group on the y-axis to compare heavy alcohol consumption across world regions:
+
+::: {.panel-tabset}
+
+# Altair
+
+Here we use `y='region'`:
+
+```{python}
+alt.Chart(income_df).mark_point().encode(
+    x='percentage_heavy_drinkers',
+    y='region',
+)
+```
+
+# ggplot
+
+Here we use `y=region`:
+
+```{r fig.width=8, fig.height=6}
+ggplot(income_df, aes(
+    x=percentage_heavy_drinkers, 
+    y=region, 
+)) + 
+geom_point()
+```
+
+:::
+
+Notice how easy it is to compare the distributions between regions?
+
+We can even take it one step further. Position can be combined with color to reinforce grouping on one variable or to introduce grouping on a second variable. 
```

Comment:
Yep I like this idea. I've moved the section below jittering. The chart is a lot more interpretable with jittering, otherwise it seems kinda pointless to group by color.